static/skin/index.js

```diff
@@ -212,18 +233,17 @@
             const filterTag = document.getElementsByName(filter)[0];
             filterTag.addEventListener('change', () => {resetAndFilter(filterTag.name, filterTag.value)});
         });
+        if (filters) {
+            window.history.pushState({}, null, `${window.location.href.split('?')[0]}?${params.toString()}`);
```

Comment:
Do we really need this ? (Or at least, do we need it all the time ?)

It seems we pass here when we have loaded the page (so after a the user arrive to the page).
So the url should already be in the history.
We may want to pushstate if we modify the "url" internally but this is not the case here.

Reply:
@mgautierfr this is needed here as in the case we get params from cookies the URL is not updated and 
L238 -> `params.forEach((value, key) => {document.getElementsByName(key)[0].value = value});`
doesn't dispatch an on change event it just changes the value if I call a dispatch event here it will mean we will be pushing into history for each separate filter so if the user wants to go back he would have to click 3-4 time which wouldn't be a good thing and as the filters were called together there should be only one step for them.

Reply:
Couldn't we dispatch a event after L238 ?
Once we have changed all the values, we "apply" them.

---

It seems to me that if push a new history state here and the user go the url with cookies, we will have two history state:
- The url without param.
- The url with param (taken from cookies).

But the two are somehow the same state has the url without param will be initialized the same way than the second (and will insert a new state).
So if the user want to go back, he would have to click 2 times (fast enough to not let the page load the fist time)

Reply:
> Couldn't we dispatch a event after L238 ?
Once we have changed all the values, we "apply" them.

we can't dispatch a single event to change the URL and apply the filters all together as it calls `resetAndFilter` which only changes URL (by pushing into history only if there is a single filter change that is passed through it).
and even if we could it would still mean we would be pushing a new state into history.

>  It seems to me that if push a new history state here and the user go the url with cookies, we will have two history state:
The url without param.
The url with param (taken from cookies).
But the two are somehow the same state has the url without param will be initialized the same way than the second (and will insert a new state).
So if the user want to go back, he would have to click 2 times (fast enough to not let the page load the fist time)

agreed upon this part but at this moment I couldn't think of something else that would have a single back click as we are changing the URL through JS as always.
this similar thing happens when there is no cookie saved and user goes to the site and browser language is detected and a filter is applied in that case as well user has to click 2 times.

Reply:
I had a look and I believe to understand the situation. On my side I requested that @MananJethwani secures that the filter status is always synced with the URL params (I believe the reasons are obvious). This forces him to add an entry in the history to set the URL params properly based on the cookies. We get then 2 entries. Obviously it is possible to to load from the cookies and don't update the URL, but then my request would not be respected anymore. A choice has to be made. In my case I strongly prefer the current situation, ie. garanty that a user can copy the URL and will get the same result again.